packages/opencensus-web-all/src/initial-load-context.ts

```diff
@@ -39,14 +40,22 @@ const DEFAULT_SAMPLE_RATE = 0.0001;
  * marked sampled.
  */
 export function getInitialLoadSpanContext(): SpanContext {
-  if (!windowWithOcwGlobals.traceparent) return randomSampledSpanContext();
-  const spanContext = traceParentToSpanContext(
-    windowWithOcwGlobals.traceparent
-  );
-  if (!spanContext) {
-    console.log(`Invalid traceparent: ${windowWithOcwGlobals.traceparent}`);
-    return randomSampledSpanContext();
+  let spanContext: SpanContext;
+  if (!windowWithOcwGlobals.traceparent) {
+    spanContext = randomSampledSpanContext();
+  } else {
+    const spanContextFromTraceparent = traceParentToSpanContext(
+      windowWithOcwGlobals.traceparent
+    );
+    if (!spanContextFromTraceparent) {
+      console.log(`Invalid traceparent: ${windowWithOcwGlobals.traceparent}`);
+      spanContext = randomSampledSpanContext();
+    } else {
+      spanContext = spanContextFromTraceparent;
+    }
   }
+  // Set the span context to allow passing it around OpenCensus Web.
+  setInitialLoadSpanContext(spanContext);
```

Comment:
What would you think about moving `setInitialLoadSpanContext` into this `initial-load-context.ts` file and making it non-exported? My thinking is that the user of the utility should just need to call `getInitialLoadSpanContext` and not need to worry about how it is generated / stored internally.

Reply:
I think having this adds more complexity and makes less understandable the code, I think that having the setter there is better. Additionally, if the `initialLoadSpanContext` will have a getter, the logic of setting this value from the `initial-load-context` is not good as we have to first get the value and then modify it.

Reply:
Final decision: Create @opencensus/web-initial-load package to store this logic rather than using the core package.